pkg/controller/job/job_controller.go

```diff
@@ -906,6 +906,11 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	needsStatusUpdate := suspendCondChanged || active != job.Status.Active || !pointer.Int32Equal(ready, job.Status.Ready)
 	job.Status.Active = active
 	job.Status.Ready = ready
+	if shouldCountTerminating(&job) {
+		terminating := pointer.Int32(controller.CountTerminatingPods(jobCtx.pods))
+		needsStatusUpdate = needsStatusUpdate || !pointer.Int32Equal(job.Status.Terminating, terminating)
+		job.Status.Terminating = terminating
+	}
```

Comment:
I think we should always assign to the field in case the feature gets disabled to reset the status field to zero.

Reply:
If the feature gate gets disabled we want it to be null not zero.

Reply:
that works as well, can you please add it there? unless I am missing something?

Reply:
It is working that way currently.  if the feature gate is off we don't see `job.Status.Terminating`.  Idk how the api update would work if I set `job.Status.Terminating=nill`.  We don't want to see terminating field if feature gate is turned off so I think guarding the update in cases when we want to count terminating is correct.

Reply:
Ik, but does this work?

1. feature gate is enabled
2. status.terminating is set to a non nil value
3. feature gate is disabled
4. status.terminating should be reset to nil (I am sorry, but not seeing the place that does that)

Reply:
I agree with @atiratree

removing `if shouldCountTerminating(&job)` should be enough to clear the field when the gate is disabled after the field was already populated.

Reply:
Pushed an update for this.